src/openrct2/scenario/ScenarioRepository.cpp

```diff
@@ -203,6 +204,19 @@ class ScenarioFileIndex final : public FileIndex<scenario_index_entry>
     }
 
 private:
+    static IStream* GetStreamFromRCT2Scenario(const std::string& path)
```

Comment:
Can you make it return a `std::unique_ptr` instead? Then we don't need to do manual memory management.

Reply:
It's what I tried initially, but that caused a crash. The same happened with shared_ptr.

Now, I'm not very knowledgeable in this area, so feel free to experiment with it yourself. If you happen to hit upon a better solution, you can push it to the PR.

Reply:
I am surprised it doesn't still crash. The problem is that your MemoryStream is still trying to access `data` which is automatically freed when `data` goes out of scope. You can tell MemoryStream that it is now the owner of the memory you pass into it, but I don't know how to take ownership away from `data`.

Reply:
The issue with this is the vector will not allow you to take ownership of the buffer. I simply copy the data into the MemoryStream now which solves this.